yt/yt/client/queue_client/consumer_client.cpp

```diff
@@ -537,12 +537,19 @@ class TGenericConsumerClient
             tabletAndRowIndices.push_back({partitionIndex, offset - 1});
         }
 
-        auto partitionRowInfos = CollectPartitionRowInfos(
-            QueueRef_->Path,
-            QueueClusterClient_,
-            std::move(tabletAndRowIndices),
-            params,
-            Logger);
+        THashMap<int, THashMap<i64, TPartitionRowInfo>> partitionRowInfos;
+        try {
+            partitionRowInfos = CollectPartitionRowInfos(
```

Comment:
I think that we generally try to avoid try-except clauses like this in our control flow. Let's make `CollectPartitionRowInfos` return a future to its result. It should be fairly easy to rewrite the logic inside into SelectRows + Apply.